flake8_fine_pytest/watchers/unique_test_names.py

```diff
@@ -0,0 +1,27 @@
+import ast
+from typing import Set
+
+from flake8_fine_pytest.watchers.base import BaseWatcher
+
+
+class UniqueTestNamesWatcher(BaseWatcher):
+    error_template = 'FP009 Duplicate name test case ({})'
+
+    def run(self) -> None:
+        testcases_names: Set[str] = set()
+
+        if not self._is_test_file(self.filename):
+            return
+
+        for node in ast.walk(self.tree):
+            if not self._is_test_function(node):
```

Comment:
Almost done :) 

Better call `_should_check_node` method in this place. This is because the should_check method tells us whether handle this node or just skip it. By default, this method checks that node is test function.
But it can be something else, for example [here](https://github.com/best-doctor/flake8-fine-pytest/blob/master/flake8_fine_pytest/watchers/xfail_until_argument_watcher.py#L69) we need more complex condition to check. So, we are overriding the default behavior and keeping the methods naming consistency.

Reply:
Done :)